src/core/actions/index.js

```diff
@@ -32,28 +47,48 @@ export function setClientApp(clientApp) {
   };
 }
 
-export function setLang(lang) {
+export type SetLangAction = {|
+  type: typeof SET_LANG,
+  payload: {| lang: string |},
+|};
+
+export function setLang(lang: string): SetLangAction {
   return {
     type: SET_LANG,
     payload: { lang },
   };
 }
 
-export function setUserAgent(userAgent) {
+export type SetUserAgentAction = {|
+  type: typeof SET_USER_AGENT,
+  payload: {| userAgent: string |},
+|};
+
+export function setUserAgent(userAgent: string): SetUserAgentAction {
   return {
     type: SET_USER_AGENT,
     payload: { userAgent },
   };
 }
 
-export function loadEntities(entities) {
+export type LoadEntitiesAction = {|
+  type: typeof ENTITIES_LOADED,
+  payload: {| entities: Array<Object> |},
+|};
+
+export function loadEntities(entities: Array<Object>): LoadEntitiesAction {
   return {
     type: ENTITIES_LOADED,
     payload: { entities },
   };
 }
 
-export function setCurrentUser(username) {
+export type SetCurrentUserAction = {|
+  type: typeof SET_CURRENT_USER,
```

Comment:
Oh, whoops, good catch. My intention here was to say the action `type` could only be equal to `'SET_CURRENT_USER'` so that we can link that in the reducer and keep everything in sync. That `typeof` call only returns the type which is `string`, like you said. Duh. I need to think about this more in https://github.com/mozilla/addons-frontend/issues/2260 . For now I just changed them all to strings.